Main.py

```diff
@@ -21,37 +21,54 @@ def App():
         for i in xrange(len(newsSources)):
             print "["+str(i)+"]" +"\t" +newsSources[i]
         print "Please enter the index of the news source or press 99 to quit"
-        try:
-            newsSourceNumber=raw_input("News Source Number >>>>")
-        except ValueError:  
-            print("That is not a valid News source number")
-        newsSourceNumber=int(newsSourceNumber)
+
+        sourceNumberOkay = False
+        while sourceNumberOkay == False:
+            try:
+                newsSourceNumber=raw_input("News Source Number >>>>")
+                newsSourceNumber=int(newsSourceNumber)
+
+                if (newsSourceNumber >=len(newsSources)):
+                    print "please select an index less than "+ str(len(newsSources))
+                else:
+                    sourceNumberOkay = True
+            except ValueError:
+                print("That is not a valid News source number")    
+
         if newsSourceNumber==99:

```

Comment:
it will never go to this place. It seems it's an infinite loop.
Best would be to put this line inside the while loop